neurogym/envs/annubes.py

```diff
@@ -138,45 +183,89 @@ def _new_trial(self, **kwargs: Any) -> dict:  # type: ignore[override]
+
+        else:
+            if self.exclusive:
+                # Mutually exclusive modalities. Pick *only* one modality.
+                stim_types = set(
+                    self._rng.choice(
+                        list(available_modalities.keys()),
+                        1,
+                        False,
+                        p=list(available_modalities.values()),
+                    )[0],
+                )
+
             else:
-                stim_type = self._rng.choice(list(self.session.keys()), p=list(self.session.values()))
-                # Update sequential count
+                # Overlap is permitted. Pick *at least* one modality.
+                stim_types = {k for k, v in available_modalities.items() if self._rng.random() <= v}
+                if len(stim_types) == 0:
+                    # Pick at least one modality from the available ones
+                    stim_types = set(
+                        self._rng.choice(
+                            list(available_modalities.keys()),
+                            1,
+                            False,
+                            list(available_modalities.values()),
+                        )[0],
+                    )
 
-            self.sequential_count = 1 if stim_type != self.last_modality else self.sequential_count + 1
-            self.last_modality = stim_type
+            self.sequential_count = {k: (v + 1 if k in stim_types else 0) for k, v in self.sequential_count.items()}
+
+            # Reset the sequential count for catch trials
+            if None in self.sequential_count:
+                self.sequential_count[None] = 0
+
+            stim_values = {k: self._rng.choice(self.stim_intensities[k], 1, False)[0] for k in stim_types}
 
-            stim_value = self._rng.choice(self.stim_intensities, 1)
             for mod in self.session:
-                if stim_type == mod:
-                    self.add_ob(stim_value, "stimulus", where=mod)
+                if mod in stim_types:
+                    self.add_ob(stim_values[mod], "stimulus", where=mod)
                     self.add_randn(0, self.noise_factor, "stimulus", where=mod)
                 self.set_groundtruth(0, period="fixation")
                 self.set_groundtruth(1, period="stimulus")
                 self.set_groundtruth(0, period="iti")
-        else:
-            self.set_groundtruth(0, period="fixation")
-            self.set_groundtruth(0, period="stimulus")
-            self.set_groundtruth(0, period="iti")
-            # Reset sequential count for catch trials
-            self.sequential_count = 1
-            self.last_modality = None
+
+        stim_types_final = list(stim_types)
 
         self.trial = {
             "catch": catch,
-            "stim_type": stim_type,
-            "stim_value": stim_value,
+            "stim_types": stim_types_final,
+            "stim_values": [stim_values[k] for k in stim_types_final],
```

Comment:
does it need to be converted to a list?
if so, wouldn't this be simpler?

```suggestion
    self.trial = {
            "catch": catch,
            "stim_types": list(stim_types),
            "stim_values": [stim_values[k] for k in stim_types],
```